lib/linter.js

```diff
@@ -19,7 +19,8 @@ class Linter {
 
   filePathTest(filePath) {
     const segments = _.compact([].concat(this.pathParser(filePath)));
-    return assertAllSegmentsPassing(segments, this.matcher);
+    if(segments.length)
```

Comment:
Yes. I encountered a situation where an empty segment would pass, it was undesirable. I am happy to change accordingly, if you have a better idea why it does.

Reply:
That makes sense. If wouldn't mind adding a test that covers this situation that would be grand.

Reply:
@pietvanzoen I am so sorry for getting back to you this late. I would really appreciate it, if you could add the required test since I don't have a clue on doing it.